board: cavs15: Add a option to control signing ways

KangJianX opened the pull request:
Fixed [#31819](https://github.com/zephyrproject-rtos/zephyr/issues/31819)

Zephyr testcases(not SOF case) not use kernel DSP driver to load image
on ADSP board, thus do not need signing with xman. So add a 'no-manifest'
to specify signing without xman in image. If use DSP driver load image,
we should not specify.
And fixed a syntax error in script when download firmware. When copy the data to buffer, the old script can not assign sequence of same size, so specify size.

Signed-off-by: Jian Kang <jianx.kang@intel.com>

lyakh commented on scripts/west_commands/sign.py:
```diff
@@ -112,6 +112,8 @@ def do_add_parser(self, parser_adder):
         group.add_argument('-D', '--tool-data', default=None,
                            help='''path to tool data/configuration directory, if needed''')
+        group.add_argument('-X', '--no-xman', action='store_true',
+                           help='''signing image without xman, if needed''')
```
maybe just `'''do not add extended manifest'''`

KangJianX replied:
Thanks, improve the description.

lyakh commented on boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:
```diff
@@ -38,7 +38,7 @@ def copy(self, data, size):
                              % (self.alloc_size, size))
         logging.debug("Copying Data to DMA buffer")
-        self.buf[:] = data[:]
+        self.buf[:size] = data[:size]
```
is this related or needed? Was the original version wrong?

KangJianX replied:
The original version wrong, it can only assign sequence of same size, buffer and data not same size, so need this change.

lyakh replied:
it changed now because you removed the manifest from the image?

KangJianX replied:
This change is not due to remove manifest. Even if add manifest, the buffer and data size still not same, it is fix for load firmware process.

lyakh replied:
ok, so it isn't really related to the main topic of this commit...

KangJianX replied:
yes, should I use a new commit for this change?

lyakh replied:
I think that would be easier to understand, yes. Thanks!

KangJianX replied:
done, do you think it's ok?

lyakh reviewed (commented):
looks good! Just a bit of clarification / clean up maybe

KangJianX committed:
board: cavs15: fixed size error that load firmware by script

When load firmware by script, the buffer size and data size are not
same, so specify size when copy data to buffer.

Signed-off-by: Jian Kang <jianx.kang@intel.com>

KangJianX committed:
board: cavs15: Add a option to control signing ways

Zephyr testcases(not SOF case) not use kernel DSP driver to load image
on ADSP board, thus do not need signing with xman. So add a input
'--no-manifest' to specify signing without xman in image. If use DSP
driver load image, we should not specify this.

Signed-off-by: Jian Kang <jianx.kang@intel.com>
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

lyakh reviewed (approved):
Thanks for addressing comments!

mbolivar-nordic reviewed (dismissed):
`--no-xman` should not be added to the common arguments, because it doesn't apply to imgtool. Please use `tool_args`.

KangJianX commented:
@mbolivar-nordic 
> `--no-xman` should not be added to the common arguments, because it doesn't apply to imgtool. Please use `tool_args`.

source code:
```
 sign_base = ([tool_path] + args.tool_args +
                     ['-o', out_bin] +  conf_path_cmd + ['-i', '3', '-e'] +
                     [bootloader, kernel])

```
From source code, the `args.tool_args` are used by imgtool already, but `--no-xman` should not be applied in imgtool, so I added a option that can be use if needed.

andyross reviewed (approved):
This seems to be at best an optimization of just a few bytes in image transferred over DMA.  The manifest doesn't go into SRAM and shouldn't change runtime behavior at all.  Not that it's a bad idea, but why do we care?  Is something breaking? +1 regardless.

That said: Marti's criticism is correct: this is a tool-specific thing with rimage and doesn't belong in the global signing arguments.

KangJianX commented:
@mbolivar-nordic address your comment, could you help to approve or add any comments?

LeiW000 commented:
I would like to see it fxied in 2.5 release. @mbolivar-nordic , please take look at the code change again.

KangJianX commented:
@mbolivar-nordic Hi, could you help to review again?

lyakh commented on boards/xtensa/intel_adsp_cavs15/tools/flash.sh:
```diff
@@ -13,7 +13,7 @@ if [ -z "$2" ]
   then
     echo "Signing with key " $key
-    west sign -d ${BUILD} -t rimage -p $4 -D $3 -- -k $2
+    west sign -d ${BUILD} -t rimage -p $4 -D $3 -- -k $2 no-manifest
```
Shouldn't this be `--no-manifest` as a proper command-line parameter?

nashif replied:
done
